mlcomp/parallelm/mlpiper/mlpiper.py

```diff
@@ -249,13 +249,18 @@ def _install_deps(self, py_deps):
 
     def run_deployment(self):
         self._logger.info("Running prepared deployment, {}".format(self._deploy_dir))
+        py_version_major = sys.version_info[0]
 
         sys.path.insert(0, self._deploy_dir)
-        for egg in glob.glob("{}/*.egg".format(self._deploy_dir)):
+        eggs = glob.glob("{}/*py{}*.egg".format(self._deploy_dir, py_version_major))
+        if len(eggs) == 0:
+            self._logger.warn("No eggs found for py{}. Trying to find all possible eggs.".format(py_version_major))
+            eggs = glob.glob("{}/*.egg".format(self._deploy_dir))
```

Comment:
Isn't it dangerous to try to pick any egg file? It might lead to errors. Can't we guess what should be there? or if we intend to rename the egg, then make sure to change the code accordingly.

Reply:
what is the general pythonic way of doing this? i am sure we arent the first to deal with multiple library versions?

Reply:
This is it.
Usually when you install egg, egg-info is installed as well. And it will be correctly installed under specific folder depending on py version. 
We don't install egg, because we just want to run a pipeline with some components, so we have to handle to add it to path. As we have multiple eggs, this is up to us to define discovery method, which is filename now, and only this one to the path.
(it's like not installing so libs)
I was not able to find more info for now.